### PR TITLE
[TestNet] Key value store contract

### PR DESCRIPTION
**Pull request by zeptin**
This is a basic contract that would typically form part of the functionality of more complex contracts.

However, it has proven useful to have a standalone contract for this on the Ethereum network.

**Compiler Version**
```
v2.0.0
```
**Contract Hash**
```
d4fedb04b097bf9056fa87f36ce5217890cee994ebe1700503be11a8bceb6872
```

**Contract Byte Code**
```
4D5A90000300000004000000FFFF0000B800000000000000400000000000000000000000000000000000000000000000000000000000000000000000800000000E1FBA0E00B409CD21B8014CCD21546869732070726F6772616D2063616E6E6F742062652072756E20696E20444F53206D6F64652E0D0D0A2400000000000000504500004C010200ACBD98F20000000000000000E00022200B013000000A00000002000000000000D2280000002000000040000000000010002000000002000004000000000000000400000000000000006000000002000000000000030040850000100000100000000010000010000000000000100000000000000000000000802800004F000000000000000000000000000000000000000000000000000000004000000C000000642800001C0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000200000080000000000000000000000082000004800000000000000000000002E74657874000000D808000000200000000A000000020000000000000000000000000000200000602E72656C6F6300000C0000000040000000020000000C0000000000000000000000000000400000420000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000B428000000000000480000000200050074210000F006000001000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000007602280500000A7201000070038C0600000104280600000A6F0700000A2A000013300400B300000001000011020314FE037211000070280800000A020414FE037237000070280800000A02036F0900000A6A0228030000066EFE0216FE017261000070280800000A02046F0900000A6A0228030000066EFE0216FE0172A3000070280800000A02280500000A720100007002280A00000A6F0B00000A8C0600000103280600000A046F0C00000A021200FE1503000002120002280A00000A6F0B00000A7D010000041200037D020000041200047D0300000406280100002B2A4602280500000A72E90000706F0E00000A2A4A02280500000A72E9000070036F0F00000A2A7A0203281000000A020416FE0372FD000070280800000A020428040000062A0042534A4201000100000000000C00000076342E302E33303331390000000005006C0000003C020000237E0000A80200003802000023537472696E677300000000E004000040010000235553002006000010000000234755494400000030060000C000000023426C6F6200000000000000020000015715A201090A000000FA013300160000010000000D000000030000000300000005000000070000001000000005000000010000000100000001000000020000000100000002000000010000000100000000005B010100000000000600AA009D010600D9009D01060096008A010F00BD0100000A000B02DC010A00CC01DC010A005E00DC0106004A006D010A00CA00DC010A008500DC01060010016D01060019026D010A003500DC01000000001500000000000100010001001000FA0100001500010001000A0110001701000021000100060006007401660006002F026A000600F7006A00502000000000860020026D000100702000000000860024023A0003002F210000000086083501740005004121000000008108430178000500542100000000861884017D00060000000100D40100000200330200000100330200000200FD0000000100FD00000001007200000002005101090084010100110084010600190084010A004900840106002900540010005900F3011500510003011C0029002802260059002A012C0029003E00300069007901350051000D013A00290026014000510001004C0051000B00510029008401570021002300B9002E000B0088002E00130091002E001B00B00041002300B9002100020001000000470184000200030003000100040003000480000000000000000000000000000000000B0200000400000000000000000000005D001E00000000000200000000000000000000000000DC0100000000030002001B004700000000000047657455496E7433320053657455496E743332003C4D6F64756C653E0053797374656D2E507269766174652E436F72654C696200494D657373616765006765745F4D6573736167650056616C756554797065006765745F53746174650049536D617274436F6E7472616374537461746500736D617274436F6E74726163745374617465004950657273697374656E7453746174650044656275676761626C6541747472696275746500436F6D70696C6174696F6E52656C61786174696F6E7341747472696275746500496E6465784174747269627574650052756E74696D65436F6D7061746962696C6974794174747269627574650056616C75650076616C756500476574537472696E6700536574537472696E67004B657956616C75654368616E6765644C6F67006765745F4C656E677468006765745F4D61784C656E677468007365745F4D61784C656E677468006D61784C656E67746800536D617274436F6E74726163742E646C6C0053797374656D0046726F6D006765745F53656E646572002E63746F720053797374656D2E446961676E6F73746963730053797374656D2E52756E74696D652E436F6D70696C6572536572766963657300446562756767696E674D6F6465730041646472657373006164647265737300537472617469732E536D617274436F6E74726163747300466F726D6174004B657956616C7565436F6E747261637400536D617274436F6E7472616374004F626A656374004765740053657400417373657274004B6579006B65790000000F7B0030007D003A007B0031007D0000254B00650079002000630061006E006E006F00740020006200650020006E0075006C006C000029560061006C00750065002000630061006E006E006F00740020006200650020006E0075006C006C0000414B006500790020006C0065006E006700740068002000630061006E006E006F007400200065007800630065006500640020006D006100780069006D0075006D000045560061006C007500650020006C0065006E006700740068002000630061006E006E006F007400200065007800630065006500640020006D006100780069006D0075006D0000134D00610078004C0065006E0067007400680000414D006100780069006D0075006D0020006C0065006E0067007400680020006D007500730074002000620065002000310020006F00720020006D006F00720065000000714B896DF5D716468D1A13CAD670299B0004200101080320000105200101111104200012290600030E0E1C1C0420010E0E040701110C05200201020E0320000804200012350420001119052002010E0E06300101011E00040A01110C042001090E052002010E0905200101121D087CEC85D7BEA7798E0306111902060E0620020E11190E03200009042001010906200201121D09032800090801000800000000001E01000100540216577261704E6F6E457863657074696F6E5468726F7773010801000200000000000401000000000000000000000000000000000010000000000000000000000000000000A82800000000000000000000C2280000002000000000000000000000000000000000000000000000B4280000000000000000000000005F436F72446C6C4D61696E006D73636F7265652E646C6C0000000000FF25002000100000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000002000000C000000D43800000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000
```

### CODE REVIEW

File: Testnet/KeyValue/KeyValue/KeyValue.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    {
+        Assert(key != null, "Key cannot be null");
+
+        Assert(value != null, "Value cannot be null");
+
+        Assert(key.Length <= this.MaxLength, "Key length cannot exceed maximum");
+
+        Assert(value.Length <= this.MaxLength, "Value length cannot exceed maximum");
+
+        State.SetString($"{Message.Sender}:{key}", value);
```
Comment (mrtpain) [inline Testnet/KeyValue/KeyValue/KeyValue.cs:21]: This is a nice to have contract, could be used for a lot of things.

Only suggestion is to add a log here to track historical changes through receipts.

**Comment by mrtpain** [comment]
Change looks good but I get a different hash of `d4fedb04b097bf9056fa87f36ce5217890cee994ebe1700503be11a8bceb6872`.

**Review by mrtpain — APPROVED** [review]
lgtm bytecode and hash match

**Comment by StratisIain** [comment]
Whitelisted

**Pull request merged**